scene/gui/code_edit.cpp

```diff
@@ -1154,6 +1154,13 @@ void CodeEdit::_new_line(bool p_split_current_line, bool p_above) {
 			set_caret_line(get_caret_line(i) - 1, false, true, 0, i);
 			set_caret_column(get_line(get_caret_line(i)).length(), i == 0, i);
 		}
+
+		// Insert a documentation comment if the caret was previously on a comment block.
+		String prev_line = get_line(get_caret_line(i) - 1);
+
+		if (prev_line.strip_edges().begins_with("##")) {
+			insert_text_at_caret("## ", i);
+		}
```

Comment:
This is a feature specific to GDScript and the Editor. I am not savvy enough to say where this should go, but definitely not in the **CodeEdit** class, which is much broader than this.

Reply:
Yes, the initial purpose of this PR was to gather information on how to do it best and to learn the codebase. @kitbdev gave me a clue about it; see https://github.com/godotengine/godot/pull/95955#issuecomment-2305904725. I can enhance it now :)

Thank you for your feedback!